packages/front/src/routes/search/[name]/Search.controller.ts

```diff
@@ -0,0 +1,24 @@
+import Store from "$lib/core/Store";
+import { returnInfinitPromise } from "$lib/helpers/promiseHelper";
+import associationService from "$lib/resources/associations/association.service";
+
+export default class SearchController {
+    associations: Store<any>;
+    searchPromise: Store<Promise<any>>;
+    inputSearch: Store<string>;
+
+    constructor(name) {
+        this.inputSearch = new Store("");
+        this.associations = new Store([]);
+        this.searchPromise = new Store(returnInfinitPromise());
+        this.searchPromise.set(this.fetchAssociationFromName(name));
+    }
+
+    fetchAssociationFromName(name) {
+        return associationService.search(name).then(associations => this.associations.set(associations));
+    }
+
+    onSubmit() {
+        this.searchPromise.set(this.fetchAssociationFromName(this.inputSearch.value));
+    }
```

Comment:
```suggestion
    onSubmit() {
        return goto(`/search/${this.inputSearch.value}`, {replaceState: true})
    }
```
Grâce à sveltekit ce code ne recharge presque rien et ça résout le pb du retour arrière. En revanche `history.replaceState` ne fonctionne pas avec sveltekit en fait

Reply:
et `goto` doit être importé : `import { goto } from "$app/navigation";`

Reply:
à première vu ne fonctionne pas tout seul ça casse le dynamisme de la page. je vois ça demain !

Reply:
chez moi c'était bon mais je te crois

Reply:
Oui en fait j'ai testé mais ça reload pas vraiment la page en effet, donc je dois trigger à la main la recherche tout de même avec 

```js
this.searchPromise.set(this.fetchAssociationFromName(this.inputSearch.value));
return goto(`/search/${this.inputSearch.value}`, { replaceState: true });
```